client/admin/interface.go

```diff
@@ -20,9 +20,34 @@
 
 package admin
 
-import "github.com/uber/cadence/.gen/go/admin/adminserviceclient"
+import (
+	"context"
+
+	"go.uber.org/yarpc"
+
+	"github.com/uber/cadence/.gen/go/admin"
+	"github.com/uber/cadence/.gen/go/replicator"
+	"github.com/uber/cadence/.gen/go/shared"
+)
 
 // Client is the interface exposed by admin service client
 type Client interface {
-	adminserviceclient.Interface
+	AddSearchAttribute(context.Context, *admin.AddSearchAttributeRequest, ...yarpc.CallOption) error
```

Comment:
qq: why we will not use the embedded thrift interface?

Reply:
This was changed on purpose so that we can change method signatures to accept and return internal types instead of thrift counterparts. Thrift types will be only be used in thrift clients and handlers. These changes will come with upcoming PRs.